packages/venia-concept/src/components/Checkout/form.js

```diff
@@ -1,329 +1,354 @@
-                        submit={this.submitPaymentMethodAndBillingAddress}
-                        submitting={submitting}
-                        countries={countries}
-                    />
-                );
-            }
-            case 'shippingMethod': {
-                const { availableShippingMethods, shippingMethod } = this.props;
-
-                return (
-                    <ShippingForm
-                        availableShippingMethods={availableShippingMethods}
-                        cancel={this.stopEditing}
-                        shippingMethod={shippingMethod}
-                        submit={this.submitShippingMethod}
-                        submitting={submitting}
-                    />
-                );
-            }
-            default: {
-                return null;
-            }
-        }
-    }
-
-    get overview() {
-        const {
-            cart,
-            classes,
-            hasPaymentMethod,
-            hasShippingAddress,
-            hasShippingMethod,
-            ready,
-            submitOrder,
-            submitting
-        } = this.props;
-
-        return (
-            <Fragment>
-                <div className={classes.body}>
-                    <Section
-                        label="Ship To"
-                        onClick={this.editAddress}
-                        showEditIcon={hasShippingAddress}
-                    >
-                        {this.shippingAddressSummary}
-                    </Section>
-                    <Section
-                        label="Pay With"
-                        onClick={this.editPaymentMethod}
-                        showEditIcon={hasPaymentMethod}
-                    >
-                        {this.paymentMethodSummary}
-                    </Section>
-                    <Section
-                        label="Use"
-                        onClick={this.editShippingMethod}
-                        showEditIcon={hasShippingMethod}
-                    >
-                        {this.shippingMethodSummary}
-                    </Section>
-                    <Section label="TOTAL">
-                        <Price
-                            currencyCode={cart.totals.quote_currency_code}
-                            value={cart.totals.subtotal || 0}
-                        />
-                        <br />
-                        <span>{cart.details.items_qty} Items</span>
-                    </Section>
-                </div>
-                <div className={classes.footer}>
-                    <Button onClick={this.dismissCheckout}>Back to Cart</Button>
-                    <SubmitButton
-                        submitting={submitting}
-                        valid={ready}
-                        submitOrder={submitOrder}
-                    />
-                </div>
-            </Fragment>
-        );
-    }
+const isSubmitDisabled = (busy, valid) => busy || !valid;
+
+const EditableForm = props => {
```

Comment:
The parent form is now split into multiple function components. Should we move these into a Form/ directory and split them out into individual files?

Reply:
@jimbo says he prefers flat structure. I like nested, for things that are specific to a parent and won't be shared. 🤷‍♂

Reply:
> The parent form is now split into multiple function components. Should we move these into a Form/ directory and split them out into individual files?

We should definitely split them out into individual files, as per existing convention. I do like the idea of a `Form/` directory personally - especially since we have a `Receipt` directory within `Checkout` already.

Reply:
> @jimbo says he prefers flat structure. I like nested, for things that are specific to a parent and won't be shared. 🤷‍♂

![fight](https://user-images.githubusercontent.com/13182778/59468074-d108d900-8df6-11e9-8442-58ba0df40685.gif)

Reply:
We should never have had a `Receipt` folder either—last year we had a few contributors who didn't like to follow conventions. 😉 

Honestly though, if you guys prefer to nest folders in places with a lot of files like this, I'm fine with it. Just don't go overboard.